### PR TITLE
VEN-1177: add ApplicationStatus.REJECTED status, set application to r…

### PR DESCRIPTION
**Pull request by haavikko**
…ejected when customer rejects order

### Closes :no_good_woman:
**[VEN-1177](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1177):**

### CODE REVIEW

File: applications/migrations/0026_auto_20210225_0850.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Generated by Django 3.1 on 2021-02-25 06:50
```
Comment (klemmari1) [inline applications/migrations/0026_auto_20210225_0850.py:1]: Would like to see the automatically generated migration filename to be renamed to something meaningful. For example `0026_add_rejected_application_status.py`.

Comment (a-rmz) [reply applications/migrations/0026_auto_20210225_0850.py:1]: Good catch @klemmari1 👍  Let's do it like that @haavikko

File: payments/models.py [hunk 2]
```diff
@@ -775,12 +775,13 @@ def update_lease_and_application(self, new_status):
...
-                application = self.lease.application
+        if application := self.lease.application:
+            if new_status == OrderStatus.REJECTED:
+                application.status = ApplicationStatus.REJECTED
+                application.save(update_fields=["status"])
+            elif new_status in OrderStatus.get_paid_statuses():
                 application.status = ApplicationStatus.HANDLED
-                self.lease.application.save(update_fields=["status"])
-
+                application.save(update_fields=["status"])
```
Comment (a-rmz) [inline payments/models.py:784]: At this point, would it make sense to extract the application logic to a util function like `get_lease_status`?

File: applications/migrations/0026_auto_20210225_0850.py (hunk 1 above)
Comment (haavikko) [reply applications/migrations/0026_auto_20210225_0850.py:1]: fixed

File: payments/models.py (hunk 2 above)
Comment (haavikko) [reply payments/models.py:784]: added get_application_status. I also moved the logic for generating sticker numbers to its own function.

**Comment by klemmari1** [comment]
This looks good to me! But perhaps I leave the approving for @a-rmz since I'm not super familiar with the project.

**Review by a-rmz — APPROVED** [review]

**Pull request merged**